autoload/test/java.vim

```diff
@@ -0,0 +1,4 @@
+let test#java#patterns = {
+  \ 'test':      ['\v^\s*public void (\w+)'],
```

Comment:
This could probably be made more specific by searching for the nearest `@Test` annotation and then finding the method name of the line below that. Suggestions on a regex to do that are welcome.